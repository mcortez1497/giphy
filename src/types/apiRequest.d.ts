interface ApiRequest {
  message?: string;
  isError: boolean;
  isLoading: boolean;
  isSuccess: boolean;
}

export { ApiRequest };
