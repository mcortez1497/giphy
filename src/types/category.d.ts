interface Category {
  _id: string;
  name: string;
}

export { Category };
