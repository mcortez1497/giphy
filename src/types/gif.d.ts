import { Category } from "types";

interface Gif {
  _id?: string;
  categories?: Category[];
  title: string;
  url: string;
}

interface GifPagination {
  count: number;
  offset: number;
  total_count: number;
}

interface GifResponse {
  images: {
    original: {
      url: string;
    };
  };
  title: string;
}

export { Gif, GifPagination, GifResponse };
