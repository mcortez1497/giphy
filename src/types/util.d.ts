interface QueryParams {
  [key: string]: string;
}

export { QueryParams }