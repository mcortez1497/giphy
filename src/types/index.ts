export * from "./category";
export * from "./gif";
export * from "./util";