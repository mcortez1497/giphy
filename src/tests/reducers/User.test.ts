import {
  setUsername,
  setUserCategories,
  setUserGifs,
  userReducer
} from "reducers";
import { Category, Gif } from "types";

const initialState = {
  username: "",
  gifs: [],
  categories: {
    items: []
  }
};

describe("User State", () => {
  describe("Actions", () => {
    it("returns setUsername action", () => {
      expect(setUsername("test")).toMatchSnapshot();
    });

    it("returns setUserCategories action", () => {
      const category = { _id: "123", name: "abc" } as Category;
      expect(setUserCategories([category])).toMatchSnapshot();
    });

    it("returns setUserGifs action", () => {
      const gif: Gif = {
        giphy_id: "giphy_123",
        fixed_url: "fixed_url",
        original_url: "original_url",
        title: "test",
        height: "100",
        categories: []
      };
      expect(setUserGifs([gif])).toMatchSnapshot();
    });
  });

  describe("Reducer", () => {
    it("add a category", () => {
      const action = {
        type: "ADD_USER_CATEGORY",
        category: { _id: "123", name: "abc" }
      };
      expect(userReducer(initialState, action)).toMatchSnapshot();
    });

    it("Adds a gif", () => {
      const action = {
        type: "ADD_USER_GIF",
        gif: {
          giphy_id: "giphy_123",
          fixed_url: "fixed_url",
          original_url: "original_url",
          title: "test",
          height: "100",
          categories: []
        }
      };
      expect(userReducer(initialState, action)).toMatchSnapshot();
    });

    it("Delete a category", () => {
      const state = {
        ...initialState,
        categories: {
          items: [{ _id: "123", name: "abc" }]
        }
      };
      const action = {
        type: "DELETE_USER_CATEGORY",
        id: "123"
      };
      expect(userReducer(state, action)).toMatchSnapshot();
    });

    it("Delete a gif", () => {
      const state = {
        ...initialState,
        gifs: [
          {
            _id: "123",
            giphy_id: "giphy_123",
            fixed_url: "fixed_url",
            original_url: "original_url",
            title: "test",
            height: "100",
            categories: []
          }
        ]
      };
      const action = {
        type: "DELETE_USER_GIF",
        id: "123"
      };
      expect(userReducer(state, action)).toMatchSnapshot();
    });

    it("Update a gif", () => {
      const state = {
        ...initialState,
        gifs: [
          {
            _id: "123",
            giphy_id: "giphy_123",
            fixed_url: "fixed_url",
            original_url: "original_url",
            title: "test",
            height: "100",
            categories: []
          }
        ]
      };
      const action = {
        type: "UPDATE_USER_GIF",
        gif: {
          _id: "123",
          giphy_id: "giphy_456",
          fixed_url: "fixed_url",
          original_url: "original_url",
          title: "test",
          height: "100",
          categories: []
        }
      };
      expect(userReducer(state, action)).toMatchSnapshot();
    });

    it("Reset User", () => {
      const state = {
        ...initialState,
        username: "test"
      };
      const action = {
        type: "RESET_USER"
      };
      expect(userReducer(state, action)).toMatchSnapshot();
    });

    it("Set Username", () => {
      const action = {
        type: "SET_USERNAME",
        username: "test"
      };
      expect(userReducer(initialState, action)).toMatchSnapshot();
    });

    it("Set User Categories", () => {
      const action = {
        type: "SET_USER_CATEGORIES",
        categories: [{ _id: "123", name: "abc" }]
      };
      expect(userReducer(initialState, action)).toMatchSnapshot();
    });

    it("Set User Gifs", () => {
      const action = {
        type: "SET_USER_GIFS",
        gifs: [
          {
            _id: "123",
            giphy_id: "giphy_123",
            fixed_url: "fixed_url",
            original_url: "original_url",
            title: "test",
            height: "100",
            categories: []
          }
        ]
      };
      expect(userReducer(initialState, action)).toMatchSnapshot();
    });
  });
});
