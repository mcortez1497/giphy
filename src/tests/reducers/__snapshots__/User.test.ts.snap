// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User State Actions returns setUserCategories action 1`] = `
Object {
  "categories": Array [
    Object {
      "_id": "123",
      "name": "abc",
    },
  ],
  "type": "SET_USER_CATEGORIES",
}
`;

exports[`User State Actions returns setUserGifs action 1`] = `
Object {
  "gifs": Array [
    Object {
      "categories": Array [],
      "fixed_url": "fixed_url",
      "giphy_id": "giphy_123",
      "height": "100",
      "original_url": "original_url",
      "title": "test",
    },
  ],
  "type": "SET_USER_GIFS",
}
`;

exports[`User State Actions returns setUsername action 1`] = `
Object {
  "type": "SET_USERNAME",
  "username": "test",
}
`;

exports[`User State Reducer Adds a gif 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [
    Object {
      "categories": Array [],
      "fixed_url": "fixed_url",
      "giphy_id": "giphy_123",
      "height": "100",
      "original_url": "original_url",
      "title": "test",
    },
  ],
  "username": "",
}
`;

exports[`User State Reducer Delete a category 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [],
  "username": "",
}
`;

exports[`User State Reducer Delete a gif 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [],
  "username": "",
}
`;

exports[`User State Reducer Reset User 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [],
  "username": "",
}
`;

exports[`User State Reducer Set User Categories 1`] = `
Object {
  "categories": Object {
    "items": Array [
      Object {
        "_id": "123",
        "name": "abc",
      },
    ],
  },
  "gifs": Array [],
  "username": "",
}
`;

exports[`User State Reducer Set User Gifs 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [
    Object {
      "_id": "123",
      "categories": Array [],
      "fixed_url": "fixed_url",
      "giphy_id": "giphy_123",
      "height": "100",
      "original_url": "original_url",
      "title": "test",
    },
  ],
  "username": "",
}
`;

exports[`User State Reducer Set Username 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [],
  "username": "test",
}
`;

exports[`User State Reducer Update a gif 1`] = `
Object {
  "categories": Object {
    "items": Array [],
  },
  "gifs": Array [
    Object {
      "_id": "123",
      "categories": Array [],
      "fixed_url": "fixed_url",
      "giphy_id": "giphy_456",
      "height": "100",
      "original_url": "original_url",
      "title": "test",
    },
  ],
  "username": "",
}
`;

exports[`User State Reducer add a category 1`] = `
Object {
  "categories": Object {
    "items": Array [
      Object {
        "_id": "123",
        "name": "abc",
      },
    ],
  },
  "gifs": Array [],
  "username": "",
}
`;
